File: app/src/main/java/com/erif/quickstate/MainActivity.java
package com.erif.quickstate;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.widget.LinearLayout;

import androidx.appcompat.app.AppCompatActivity;
import androidx.coordinatorlayout.widget.CoordinatorLayout;

public class MainActivity extends AppCompatActivity {

    private QuickState state;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        CoordinatorLayout parentView = findViewById(R.id.act_main_parentLayout);
        state = new QuickState(parentView, true);
        LinearLayout contentLoader = findViewById(R.id.loadingLayout);

        loadingProcess();
        state.contentLoader(contentLoader);
        state.onClickListener((stateView, buttonState, tag) -> {
            if (buttonState == QuickState.BUTTON_SINGLE) {
                Intent intent = new Intent(MainActivity.this, MainActivity2.class);
                startActivity(intent);
            }
        });
    }

    private void loadingProcess() {
        Handler handler = new Handler();
        Runnable run = () -> state.show(ConstantState.EMPTY);
        handler.postDelayed(run, 1500);
    }

}

File: app/src/main/java/com/erif/quickstates/MainActivity.kt
package com.erif.quickstates

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.erif.quickstates.examples.ActAnimated
import com.erif.quickstates.examples.ActDefault
import com.erif.quickstates.examples.ActFonts
import com.erif.quickstates.examples.ActFullState
import com.erif.quickstates.examples.ActImage
import com.google.android.material.button.MaterialButton

class MainActivity: AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.act_main)

        intent(R.id.act_main_btnDefault, ActDefault::class.java)
        intent(R.id.act_main_btnFont, ActFonts::class.java)
        intent(R.id.act_main_btnIllustration, ActImage::class.java)
        intent(R.id.act_main_btnAnimated, ActAnimated::class.java)
        intent(R.id.act_main_btnFull, ActFullState::class.java)

    }

    private fun intent(id: Int, destination: Class<*>) {
        val btn: MaterialButton = findViewById(id)
        btn.setOnClickListener {
            val intent = Intent(this, destination)
            startActivity(intent)
        }
    }

}